composeApp/src/commonMain/kotlin/com/github/familyvault/services/ChatService.kt

```diff
@@ -211,6 +210,38 @@ class ChatService(
         }
     }
 
+    override suspend fun retrievePublicKeysOfChatThreadManagers(threadId: String): List<String> {
+        val threadItem = privMxClient.retrieveThread(threadId)
+
+        return threadItem.managers
+    }
+
+    override suspend fun retrieveChatThreadInitialManagers(threadId: String): List<String> {
+        val threadItem = privMxClient.retrieveThread(threadId)
+
+        return threadItem.privateMeta.initialManagersPublicKeys ?: listOf()
+    }
+
+    override suspend fun updateGroupChatThreadsAfterUserPermissionChange(updatedUser: FamilyMember, familyMembers: List<FamilyMember>) {
+        val threadsList = retrieveAllChatThreads().filter { it.type == ChatThreadType.GROUP }
+
+        if (updatedUser.permissionGroup == FamilyGroupMemberPermissionGroup.Guardian) {
+            for (thread in threadsList) {
+                if (updatedUser.id in thread.participantsIds) {
+                    updateChatThread(thread, familyMembers, null, updatedUser)
```

Comment:
Mogę tutaj nie rozumieć, ale czy jak zmienasz uprawnienia danego usera to ustawiasz go jako stworzyciela grupy? On powinien zostać tylko managerem.

Jeżeli jest tutaj coś innego to parametry `updateChatThread` są strasznie nieczytelne.